kuwala/core/backend/app/controller/data_source/data_source.py

```diff
@@ -20,7 +21,9 @@ def get_data_source_and_data_catalog_item_id(
             status_code=404, detail=f"No data source found with ID {data_source_id}."
         )
 
-    if not data_catalog_item_id or data_catalog_item_id != "postgres":
+    if not data_catalog_item_id or (
+        data_catalog_item_id != "postgres" and data_catalog_item_id != "bigquery"
+    ):
```

Comment:
could be simplified into 

```
items = ['postgres', 'bigquery']
if not datacatalog_item_id in items
...
```

I think this will be more maintainable in the long term